presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java

```diff
@@ -322,8 +322,13 @@ public void onSuccess(PagesResponse result)
                         }
 
                         if (!isNullOrEmpty(taskInstanceId) && !result.getTaskInstanceId().equals(taskInstanceId)) {
-                            // TODO: update error message
-                            throw new PrestoException(REMOTE_TASK_MISMATCH, format("%s (%s)", REMOTE_TASK_MISMATCH_ERROR, fromUri(uri)));
+                            String msg = format("%s (%s).\nExpected TaskInstanceId: %s, received TaskInstanceId: %s",
```

Comment:
You can safely inline this. Also, avoid `\n` in exc msg (there are some exceptions, but here no need):

```java
throw new PrestoException(REMOTE_TASK_MISMATCH, format(
        "%s (%s). Expected taskInstanceId: '%s', received taskInstanceId: '%s'",
        REMOTE_TASK_MISMATCH_ERROR,
        fromUri(uri),
        taskInstanceId,
        result.getTaskInstanceId()));
```

Reply:
Got it.